ndbench-dynamodb-plugins/src/main/java/com/netflix/ndbench/plugin/dynamodb/DynamoDBTransactions.java

```diff
@@ -0,0 +1,144 @@
+        String partitionKeyName = config.getAttributeName();
+        long rcu = Long.parseLong(config.getReadCapacityUnits());
+        long wcu = Long.parseLong(config.getWriteCapacityUnits());
+
+        this.deleteTables = new ArrayList<>();
+        // create main table object
+        CreateDynamoDBTable mainTable = new CreateDynamoDBTable(dynamoDB, tableName, partitionKeyName, rcu, wcu);
+        this.deleteTables.add(new DeleteDynamoDBTable(dynamoDB, tableName, partitionKeyName));
+
+        // create child tables objects
+        ArrayList<CreateDynamoDBTable> childTables = new ArrayList<>();
+        for (int i = 0; i < config.getMainTableColsPerRow(); i++)
+        {
+            String childTableName = config.getChildTableNamePrefix() + i;
+            childTables.add(new CreateDynamoDBTable(dynamoDB, childTableName, partitionKeyName, rcu, wcu));
+            deleteTables.add(new DeleteDynamoDBTable(dynamoDB, childTableName, partitionKeyName));
+        }
+
+        // execute main table creation
+        logger.info("Creating main table programmatically");
+        TableDescription td = mainTable.get();
+        logger.info("Main Table Description: " + td.toString());
+
+        // execute child tables creation
+        logger.info("Creating child table programmatically");
+        for (int i = 0; i < config.getMainTableColsPerRow(); i++)
+        {
+            td = childTables.get(i).get();
+            logger.info("Child Table Description: " + td.toString());
+        }
+
+        DescribeLimits describeLimits = new DescribeLimits(dynamoDB, tableName, partitionKeyName);
+        final DescribeLimitsResult limits = describeLimits.get();
+        if (config.getAutoscaling()) {
+            // main table
+            dynamoDBAutoscalingConfigurer.setupAutoscaling(rcu, wcu, config.getTableName(), limits,
+                    Integer.valueOf(config.getTargetWriteUtilization()),
+                    Integer.valueOf(config.getTargetReadUtilization()));
+
+            // child tables
+            for (int i = 0; i < config.getMainTableColsPerRow(); i++)
+            {
+                dynamoDBAutoscalingConfigurer.setupAutoscaling(rcu, wcu, config.getChildTableNamePrefix() + i, limits,
+                                                               Integer.valueOf(config.getTargetWriteUtilization()),
+                                                               Integer.valueOf(config.getTargetReadUtilization()));
+            }
+        }
+    }
+
+    @Override
+    public String readSingle(String key)
+    {
+        // read from main table
+        return singleRead.apply(key);
+    }
+
+    /**
+     * Executes a write transaction, with writes spanning main table and multiple
+     * child tables indicated by colsPerRow config property
+     * @param key
+     * @return
+     */
+    @Override
+    public String writeSingle(String key)
+    {
+        return transactionWrite.apply(key);
+    }
+
+    @Override
```

Comment:
I have found out that keeping the NDBench plugin not doing the delete table is better because you have to recreate it with all the needed properties every time you start restart it. Secondly, if you deploy NDBench in multiple nodes, then all of them will try to delete the table and most of them will fail. Hence, if the plugin is simple, you can do all the table admin operations on the console which is much easier.

Reply:
@ipapapa submitted the change. let me know.